### PR TITLE
test(targets): Add itest for targets POST and DELETE

### PR DESCRIPTION
**Pull request by jan-law**
Related #474 

Tests the `TargetsPostHandler` and `TargetDeleteHandler`.

### CODE REVIEW

File: src/test/java/itest/TargetPostDeleteIT.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    @Test
+    public void testDeleteTargetThrowsWithNonExistentConnectUrl() throws Exception {
+
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        final String nonExistentUrl =
+                URLEncodedUtils.formatSegments(
+                        String.format("service:jmx:rmi:///jndi/rmi://invalid:9091/jmxrmi"));
```
Comment (jan-law) [inline src/test/java/itest/TargetPostDeleteIT.java:213]: I keep getting a `400` when trying different variations of JMX service URLs or `hostname:port`. How can I trigger a `404`?

Comment (jan-law) [reply src/test/java/itest/TargetPostDeleteIT.java:213]: Relevant code in `TargetDeleteHandler`:
```
String targetId = params.getPathParams().get("targetId");
            if (StringUtils.isBlank(targetId)) {
                throw new ApiException(400, "Invalid targetId");
            }
            URI uri = URIUtil.createAbsolute(targetId);
            if (!customTargetPlatformClient.removeTarget(uri)) {
                throw new ApiException(404);
            }
```

Comment (andrewazores) [reply src/test/java/itest/TargetPostDeleteIT.java:213]: I think the issue is that you're POSTing a form with the targetId - it should be a DELETE verb, and the request path should include the targetId as a path parameter.

Comment (jan-law) [reply src/test/java/itest/TargetPostDeleteIT.java:210]: Whoops, thanks

**Review by andrewazores — COMMENTED** [review]
Looks good, just needs a rebase.

**Review by andrewazores — APPROVED** [review]

**Pull request merged**